src/main/java/com/insidious/plugin/ui/testrunnerinjection/util/RunnerWriter.java

```diff
@@ -0,0 +1,133 @@
+            InsidiousNotification.notifyMessage("Runner file for the module: "+moduleName+", already exists at: "+ runnerFile.getAbsolutePath(), NotificationType.WARNING);
+            return;
+        }
+
+        // Write the contents to the runner file
+        writeContentsInFile(runnerFile);
+
+        // Refresh the Virtual File System to reflect the new file
+        VirtualFileManager.getInstance()
+                .refreshAndFindFileByUrl(FileSystems.getDefault().getPath(runnerFile.getAbsolutePath()).toUri().toString());
+    }
+
+    /**
+     * Writes the generated content into the specified file.
+     *
+     * @param runnerFile the file to write the contents into
+     */
+    private void writeContentsInFile(File runnerFile) {
+        String runnerFileContent = generateRunnerFileContent();
+        try (FileOutputStream fileOutputStream = new FileOutputStream(runnerFile)) {
+            fileOutputStream.write(runnerFileContent.getBytes(StandardCharsets.UTF_8));
+        } catch (IOException e) {
+            InsidiousNotification.notifyMessage("Failed to write runner file: " + e.getMessage(), NotificationType.ERROR);
+        }
+    }
+
+    /**
+     * Retrieves a module by its name.
+     *
+     * @param moduleName the name of the module to retrieve
+     * @return the Module object, or null if not found
+     */
+    private Module getModuleByName(String moduleName) {
+        ModuleManager moduleManager = ModuleManager.getInstance(project);
+        Module[] modules = moduleManager.getModules();
+        return Arrays.stream(modules)
+                .filter(module -> module.getName().equals(moduleName))
+                .findFirst()
+                .orElse(null);
+    }
+
+    /**
+     * Constructs the file for UnloggedTestRunner based on the module path.
+     *
+     * @param moduleDir the path of the module directory
+     * @return the File object representing the UnloggedTestRunner file.
+     */
+    private File constructTestRunnerFile(VirtualFile moduleDir) {
+
+        String desiredDirectoryPath;
+
+        if(moduleDir == null) {
+            desiredDirectoryPath = project.getBasePath() + testFileDir;
+        } else {
+            desiredDirectoryPath = moduleDir.getPath() + testFileDir;
+        }
+        File desiredDirectory = new File(desiredDirectoryPath);
+        desiredDirectory.mkdirs();
+        String desiredPath = desiredDirectoryPath + separator + testFileName;
+        File runnerFile = new File(desiredPath);
+        return runnerFile;
+    }
+
+    /**
+     * Generates the content for the test runner file.
+     *
+     * @return the content string for the test runner file
+     */
+    private String generateRunnerFileContent() {
+        return "import io.unlogged.runner.UnloggedTestRunner;\n" +
+                "import org.junit.runner.RunWith;\n\n" +
+                "@RunWith(UnloggedTestRunner.class)\n" +
+                "public class UnloggedTest {\n" +
+                "}\n";
+    }
```

Comment:
This code is present in 2 places, here and `TestRunnerInjector`. They could be managed from a single source.

Reply:
will take this up with other changes in the revision

Reply:
created CommonUtil class that will have a function generateRunnerFileContent()